apps/aehttp/test/aehttp_integration_SUITE.erl

```diff
@@ -906,10 +922,19 @@ block_txs_count_pending(_Config) ->
     rpc(aec_conductor, start_mining, []),
     timer:sleep(100),% so the miner is started
     add_spend_txs(),
-    {ok, PendingBlock} = get_pending_block(),
-    TxsCount = length(aec_blocks:txs(PendingBlock)), 
-    {ok, 200, #{<<"count">> := TxsCount}} =
-        get_block_txs_count_preset("pending"),
+    GetPending =
+        fun TryToGet(Attempts) when Attempts < 1 ->
+                {error, no_pending_block_returned};
+            TryToGet(Attempts) ->
+                case get_block_txs_count_preset("pending") of
+                    {ok, 200, P} -> {ok, P};
+                    {ok, 404, _} -> TryToGet(Attempts -1)
+                end
+        end,
+    {ok, Txs} = rpc(aec_tx_pool, peek, [infinity]),
+    ct:log("Txs count in pool: ~p", [length(Txs)]),
+    {ok, #{<<"count">> := TxsCount}} = GetPending(10),
```

Comment:
Reuse aec_test_utils:wait_for_it?

Reply:
I see [CI error](https://travis-ci.org/aeternity/epoch/jobs/329058030#L1487):
```
%%% aehttp_integration_SUITE ==> all_endpoints.internal_endpoints.block_txs_count_pending: FAILED
%%% aehttp_integration_SUITE ==> {{badmatch,{error,no_pending_block_returned}},
 [{aehttp_integration_SUITE,block_txs_count_pending,1,
                            [{file,"/home/travis/build/aeternity/epoch/_build/test/lib/aehttp/test/aehttp_integration_SUITE.erl"},
                             {line,936}]},
  {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1529}]},
  {test_server,run_test_case_eval1,6,[{file,"test_server.erl"},{line,1045}]},
  {test_server,run_test_case_eval,9,[{file,"test_server.erl"},{line,977}]}]}
```

I think it is because of lack of sleep (in test I mean).

Reply:
We don't actually know in advance the count of transactions in the block candidate (esp. this is prone to races) and thus we can not use aec_test_utils:wait_for_it/2,3

Reply:
You can. You just have to be a bit creative.
Set the value to true and write the fun so that it only evaluates to true when your condition is met.